fetch.bs

```diff
@@ -4648,21 +4648,102 @@ steps:
       <p class=note>The `<code>GET</code>` <a for=/>method</a> restriction serves no useful purpose
       other than being interoperable.
 
+     <li><p>Set <var>response</var> to a new <a for=/>response</a>.
+
+     <li><p>If <var>request</var>'s <a for=response>header list</a> contains `<code>Range</code>`,
+     set <var>response</var>'s <a for=response>range-requested flag</a>.
+
      <li><p>Let <var>bodyWithType</var> be the result of <a for=BodyInit>safely extracting</a>
      <var>blobURLEntry</var>'s <a for="blob URL entry">object</a>.
 
      <li><p>Let <var>body</var> be <var>bodyWithType</var>'s <a for="body with type">body</a>.
 
-     <li><p>Let <var>length</var> be <var>body</var>'s <a for=body>length</a>,
+     <li><p>Let <var>fullLength</var> be <var>body</var>'s <a for=body>length</a>,
      <a lt="serialize an integer">serialized</a> and <a>isomorphic encoded</a>.
 
      <li><p>Let <var>type</var> be <var>bodyWithType</var>'s <a for="body with type">type</a> if it
      is non-null; otherwise the empty byte sequence.
 
-     <li><p>Return a new <a for=/>response</a> whose <a for=response>status message</a> is
-     `<code>OK</code>`, <a for=response>header list</a> is « (`<code>Content-Length</code>`,
-     <var>length</var>), (`<code>Content-Type</code>`, <var>type</var>) », and
-     <a for=response>body</a> is <var>body</var>.
+     <li>
+      <p>If <var>response</var>'s <a for=response>range-requested flag</a> is unset, run the
+      following steps:
+
+      <ol>
+       <li><p>Set <var>response</var>'s <a for=response>status message</a> to `<code>OK</code>`.
+
+       <li><p>Set <var>response</var>'s <a for=response>body</a> to <var>body</var>.
+
+       <li><p>Set <var>response</var>'s <a for=response>header list</a> to «
+       (`<code>Content-Length</code>`, <var>fullLength</var>), (`<code>Content-Type</code>`,
+       <var>type</var>) ».
+      </ol>
+
+     <li>
+      <p>Otherwise, if <var>response</var>'s <a for=response>range-requested flag</a> is set, run the
+      following steps:
```

Comment:
```suggestion
      <p>Otherwise:
```
(I was initially thinking we could maybe keep it without the "if" or have it as assert, but since there's only two branches I don't think that's warranted.)